Samples/Client/SwiftClient/SwiftClientTests/SwiftClientTests.swift

```diff
@@ -0,0 +1,270 @@
+import XCTest
+@testable import SwiftClient
+import WebRTC
+
+class SwiftClientTests: XCTestCase {
+    var videoStreamVC: VideoStreamViewController!
+    var timeoutSecs: TimeInterval! = TimeInterval(50)
+    let dispatchGroup = DispatchGroup() //Force only one test to run at a time
+    let randomPeerId = 10000
+    
+    override func setUp() {
+        super.setUp()
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        videoStreamVC = storyboard.instantiateViewController(withIdentifier: "VideoStreamViewController") as! VideoStreamViewController
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        videoStreamVC.disconnectAsync(completionHandler: nil)
+        videoStreamVC = nil
+        timeoutSecs = nil
+    }
+    
+    func testSignInAsync() {
+        dispatchGroup.enter()
+        let signInExpectation = expectation(description: "signInExpectation")
+        self.videoStreamVC.signInAsync { (data, response, error) in
+            XCTAssertNil(error)
+            XCTAssertNotNil(data)
+            XCTAssertNotNil(response)
+            XCTAssertEqual(response!.statusCode, 200)
+            self.videoStreamVC.myId = 1000 // populate to avoid crashing from not calling connectAsync
+            signInExpectation.fulfill()
+            self.dispatchGroup.leave()
+        }
+        self.dispatchGroup.wait()
+        waitForExpectations(timeout: timeoutSecs, handler: nil)
+    }
+    
+    func testConnectAsync() {
+        dispatchGroup.enter()
+        let connectAsyncExpectation = expectation(description: "connectAsyncExpectation")
+        videoStreamVC.connectAsync { (error) in
```

Comment:
what does this simulate connecting to?

Reply:
This tests the function connectAsync, which is an abstraction of a lot of our flow (it signs in, starts heartbeat, hanging get, etc) and if there is a problem it disconnects. There are more granular tests that test specifically things like signing in, but the idea here is to test them all together